app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_send_feedback:
+                Intent i = new Intent(Intent.ACTION_SEND);
+                Feedback fb = new Feedback();
+                fb.sendFeedback(i);
+                startActivity(Intent.createChooser(i, "Send mail..."));
+                return true;
+            case R.id.menu_item_delete_todo:
+                UUID todoId = mTodo.getID();
+                TodoLab.get(getActivity()).deleteTodoItem(todoId);
+
+                String titleToDisplay;
+                if (mTodo.getTitle() == null) {
+                    titleToDisplay = "Unnamed task";
+                } else if (mTodo.getTitle().length() > 25) {
+                    titleToDisplay = "\"" + mTodo.getTitle().trim().substring(0, 25) + "\"";
+                } else {
+                    titleToDisplay = "\"" + mTodo.getTitle().trim() + "\"";
+                }
+
+                Toast.makeText(getActivity(),
+                        titleToDisplay + " deleted",
+                        Toast.LENGTH_SHORT).show();
+                getActivity().finish();
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        saveNewItem();
+        Log.v("DETAIL_PAUSE", mTodo.getID().toString());
+        super.onPause();
+    }
+
+    private void saveNewItem() {
+        if (mTodo.getTitle() == null) {
+            UUID mTodoID = mTodo.getID();
+            TodoLab.get(getActivity()).deleteTodoItem(mTodoID);
+        }
+
+        TodoLab.get(getActivity()).updateItem(mTodo);
+        Log.v("DETAIL_SAVE", mTodo.getID().toString());
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
+            savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_todo, container, false);
+        mTitle = (EditText) v.findViewById(R.id.todo_title);
+
+        if (mTodo.isCompleted()) {
+            mTitle.setTextColor(getResources().getColor(R.color.inactiveText));
+        }
+
+        mTitle.setFocusable(true);
+        mTitle.requestFocus();
+
+        mTitle.setText(mTodo.getTitle());
+        mTitle.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                mTodo.setTitle(s.toString());
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+
+        mDueDateTextField = (TextView) v.findViewById(R.id.todo_date);
+        updateDate();
+
+        mDateComponents = (LinearLayout) v.findViewById(R.id.date_components);
```

Comment:
I would organize all of the view variable initializations into one place so it's easier to read everything else that is going on